conan/tools/env/environment.py

```diff
@@ -500,20 +541,28 @@ def save_script(self, filename):
         if ext:
             is_bat = ext == ".bat"
             is_ps1 = ext == ".ps1"
+            is_fish = ext == ".fish"
         else:  # Need to deduce it automatically
             is_bat = self._subsystem == WINDOWS
             is_ps1 = self._conanfile.conf.get("tools.env.virtualenv:powershell", check_type=bool)
+            is_fish = self._conanfile.conf.get("tools.env.virtualenv:fish", check_type=bool)
             if is_ps1:
                 filename = filename + ".ps1"
                 is_bat = False
+            elif is_bat:
+                filename = filename + ".bat"
+            elif is_fish:
+                filename = filename + ".fish"
             else:
-                filename = filename + (".bat" if is_bat else ".sh")
+                filename = filename + ".sh"
 
         path = os.path.join(self._conanfile.generators_folder, filename)
         if is_bat:
             self.save_bat(path)
         elif is_ps1:
             self.save_ps1(path)
+        elif is_fish:
```

Comment:
as number of values grows here, it keeps adding more and more chained conditions.
some sort of generalization would be nice to apply (at least map `format` -> `save_func`, or maybe as a new parameterized function).

Reply:
I'm all for clean code, and if any more are added in the future, that'd be probably the right call :)

But right now I think that `save_script` is the perfect place to have this logic and an if with 3 branches is the easiest solution